README.md

```diff
@@ -84,6 +105,12 @@ The `$SSH_AGENT_PID` contains the process id of the agent. This is used to kill
 
 Since each job [runs in a fresh instance](https://help.github.com/en/articles/about-github-actions#job) of the virtual environment, the SSH key will only be available in the job where this action has been referenced. You can, of course, add the action in multiple jobs or even workflows. All instances can use the same `SSH_PRIVATE_KEY` secret.
 
+### Cannot Precede the `checkout` Action
+
+The `ssh-agent` step *cannot* precede the `checkout` step, though. The `checkout` action undoes the effects of `ssh-agent`. This will cause errors like:
+
+    ssh: Could not resolve hostname key-<hex-key-id>.github.com: Name or service not known
```

Comment:
Once https://github.com/actions/checkout/pull/833 is merged, that should not happen unless submodules are turned on?

Reply:
I think this is linked to #148 which is caused by the [checkout action setting up a temp HOME directory](https://github.com/actions/checkout/blob/v3.1.0/src/git-auth-helper.ts#L122) which breaks the ssh configuration.
On linux, submodule support is OK with `ssh-agent` preceding the `checkout` action.